library/core/src/ptr/metadata.rs

```diff
@@ -194,13 +204,25 @@ impl<Dyn: ?Sized> DynMetadata<Dyn> {
     /// Returns the size of the type associated with this vtable.
     #[inline]
     pub fn size_of(self) -> usize {
-        self.vtable_ptr.size_of
+        #[cfg(bootstrap)]
+        return self.vtable_ptr.size_of;
+        #[cfg(not(bootstrap))]
+        // SAFETY: DynMetadata always contains a valid vtable pointer
+        return unsafe {
+            crate::intrinsics::vtable_size(self.vtable_ptr as *const VTable as *const ())
```

Comment:
Since the intrinsic is only used for this specific use case, maybe make the intrinsic take a `&'static VTable` argument?

Reply:
That's possible, but then the type has to be available in `intrinsics.rs`, which didn't seem great? I think it would also force the type to be `pub`.

Reply:
Hmm... yea, none of this is ideal. I guess we'll go with what you have for now